File: backend/app/schemas/resume.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ResumeCreate(BaseModel):
    title: str
    desired_position: str
    about: str | None = None
    city: str | None = None
    salary_expectation: int | None = None


class ResumeUpdate(BaseModel):
    title: str | None = None
    desired_position: str | None = None
    about: str | None = None
    city: str | None = None
    salary_expectation: int | None = None
    is_active: bool | None = None


class ResumeResponse(BaseModel):
    id: int

    user_id: int

    title: str
    desired_position: str

    about: str | None
    city: str | None

    salary_expectation: int | None

    is_active: bool

    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True,
    )

File: frontend/types/resume.ts
export interface Resume {
    id: number;

    user_id: number;

    title: string;

    desired_position: string;

    about: string | null;

    city: string | null;

    salary_expectation: number | null;

    is_active: boolean;

    created_at: string;

    updated_at: string;
}


export interface ResumeCreate {
    title: string;

    desired_position: string;

    about?: string | null;

    city?: string | null;

    salary_expectation?: number | null;
}


export interface ResumeUpdate {
    title?: string;

    desired_position?: string;

    about?: string | null;

    city?: string | null;

    salary_expectation?: number | null;

    is_active?: boolean;
}